packages/mui-component-mapper/src/files/field-array.js

```diff
@@ -0,0 +1,248 @@
+    buttonLabels,
+    ...rest
+  } = useFieldApi(props);
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const combinedButtonLabels = {
+    ...defaultButtonLabels,
+    ...buttonLabels
+  };
+
+  const classes = useFielArrayStyles();
+
+  const { dirty, submitFailed, error } = meta;
+  const isError = (dirty || submitFailed) && error && typeof error === 'string';
+
+  return (
+    <FormFieldGrid {...FormFieldGridProps} className={clsx(classes.fieldArrayGroup, FormFieldGridProps.classname)}>
+      <FormControl component="fieldset" error={isError} {...FormControlProps} className={clsx(classes.formControl, FormControlProps.className)}>
+        <FieldArray key={rest.input.name} name={rest.input.name} validate={arrayValidator}>
+          {({ fields: { map, value = [], push, remove } }) => {
+            const pushWrapper = () => {
+              dispatch({ type: 'resetHistory' });
+              push(defaultItem);
+            };
+
+            const removeWrapper = (index) => {
+              dispatch({ type: 'action', action: { action: 'remove', value: value[index] } });
+              remove(index);
+            };
+
+            const undo = () => {
+              push(state.history[state.index - 1].value);
+
+              dispatch({ type: 'undo' });
+            };
+
+            const redo = () => {
+              remove(value.length - 1);
+
+              dispatch({ type: 'redo' });
+            };
+
+            return (
+              <Grid container spacing={3}>
+                <Grid item xs={12} className={classes.header}>
+                  {label && (
+                    <Typography variant="h6" className={classes.label}>
+                      {label}
+                    </Typography>
+                  )}
+                  <IconButton color="primary" aria-label="undo" component="span" disabled={state.index === 0} onClick={undo}>
+                    <UndoIcon />
+                  </IconButton>
+                  <IconButton color="primary" aria-label="redo" component="span" disabled={state.index === state.history.length} onClick={redo}>
+                    <RedoIcon />
+                  </IconButton>
+                  <Button color="primary" onClick={pushWrapper} disabled={value.length >= maxItems}>
+                    {combinedButtonLabels.add}
+                  </Button>
+                </Grid>
+                {description && (
+                  <Grid item xs={12}>
+                    <Typography variant="subtitle1">{description}</Typography>
+                  </Grid>
+                )}
+                <Grid item xs={12}>
+                  {value.length <= 0 ? (
+                    <Typography variant="body1" gutterBottom className={classes.centerText}>
+                      {noItemsMessage}
+                    </Typography>
+                  ) : (
+                    map((name, index) => (
+                      <ArrayItem
+                        key={`${name}-${index}`}
```

Comment:
I dont think that we need the `-${index}` suffix. The name from the map function should be always unique.